dev/src/expression.ts

```diff
@@ -0,0 +1,1425 @@
+  }
+
+  length(): Length {
+    return new Length(this);
+  }
+
+  like(pattern: string): Like;
+  like(pattern: Expr): Like;
+  like(stringOrExpr: string | Expr): Like {
+    if (stringOrExpr instanceof Expr) {
+      return new Like(this, stringOrExpr);
+    }
+    return new Like(this, Constant.of(stringOrExpr));
+  }
+
+  regexContains(pattern: string): RegexContains;
+  regexContains(pattern: Expr): RegexContains;
+  regexContains(stringOrExpr: string | Expr): RegexContains {
+    if (stringOrExpr instanceof Expr) {
+      return new RegexContains(this, stringOrExpr);
+    }
+    return new RegexContains(this, Constant.of(stringOrExpr));
+  }
+
+  regexMatch(pattern: string): RegexMatch;
+  regexMatch(pattern: Expr): RegexMatch;
+  regexMatch(stringOrExpr: string | Expr): RegexMatch {
+    if (stringOrExpr instanceof Expr) {
+      return new RegexMatch(this, stringOrExpr);
+    }
+    return new RegexMatch(this, Constant.of(stringOrExpr));
+  }
+
+  startsWith(prefix: string): StartsWith;
+  startsWith(prefix: Expr): StartsWith;
+  startsWith(stringOrExpr: string | Expr): StartsWith {
+    if (stringOrExpr instanceof Expr) {
+      return new StartsWith(this, stringOrExpr);
+    }
+    return new StartsWith(this, Constant.of(stringOrExpr));
+  }
+
+  endsWith(suffix: string): EndsWith;
+  endsWith(suffix: Expr): EndsWith;
+  endsWith(stringOrExpr: string | Expr): EndsWith {
+    if (stringOrExpr instanceof Expr) {
+      return new EndsWith(this, stringOrExpr);
+    }
+    return new EndsWith(this, Constant.of(stringOrExpr));
+  }
+
+  toLowercase(): ToLowercase {
+    return new ToLowercase(this);
+  }
+
+  toUppercase(): ToUppercase {
+    return new ToUppercase(this);
+  }
+
+  trim(): Trim {
+    return new Trim(this);
+  }
+
+  strConcat(...elements: (string | Expr)[]): StrConcat {
+    const exprs = elements.map(e => (e instanceof Expr ? e : Constant.of(e)));
+    return new StrConcat(this, exprs);
+  }
+
+  mapGet(subfield: string): MapGet {
+    return new MapGet(this, subfield);
+  }
+
+  count(): Count {
+    return new Count(this, false);
+  }
+
+  sum(): Sum {
+    return new Sum(this, false);
+  }
+
+  avg(): Avg {
+    return new Avg(this, false);
+  }
+
+  min(): Min {
+    return new Min(this, false);
+  }
+
+  max(): Max {
+    return new Max(this, false);
+  }
+
+  cosineDistance(other: Expr): CosineDistance;
+  cosineDistance(other: VectorValue): CosineDistance;
+  cosineDistance(other: number[]): CosineDistance;
+  cosineDistance(other: Expr | VectorValue | number[]): CosineDistance {
+    if (other instanceof Expr) {
+      return new CosineDistance(this, other);
+    } else {
+      return new CosineDistance(this, Constant.ofVector(other));
+    }
+  }
+
+  dotProductDistance(other: Expr): DotProductDistance;
+  dotProductDistance(other: VectorValue): DotProductDistance;
+  dotProductDistance(other: number[]): DotProductDistance;
+  dotProductDistance(other: Expr | VectorValue | number[]): DotProductDistance {
```

Comment:
Might be more accurate to just call this this and the class `dotProduct`. Drop "Distance"

Reply:
Will do.